common/src/main/java/com/wynntils/commands/OnlineMembersCommand.java

```diff
@@ -52,91 +49,57 @@ public LiteralArgumentBuilder<CommandSourceStack> getCommandBuilder(
 
     private int lookupGuild(CommandContext<CommandSourceStack> context) {
         CompletableFuture.runAsync(() -> {
-            CompletableFuture<MutableComponent> completableFuture =
-                    getGuildJson(context.getArgument("guildName", String.class));
+            CompletableFuture<Guild> completableFuture =
+                    Models.Guild.getGuild(context.getArgument("guildName", String.class));
 
-            completableFuture.whenComplete((result, throwable) -> McUtils.sendMessageToClient(result));
-        });
-
-        context.getSource()
-                .sendSuccess(
-                        () -> Component.translatable("command.wynntils.onlineMembers.lookingUp")
-                                .withStyle(ChatFormatting.GREEN),
-                        false);
-
-        return 1;
-    }
-
-    private static CompletableFuture<MutableComponent> getGuildJson(String inputName) {
-        CompletableFuture<MutableComponent> future = new CompletableFuture<>();
-
-        String guildName = Models.Guild.getGuildNameFromString(inputName);
-
-        ApiResponse apiResponse = Managers.Net.callApi(UrlId.DATA_WYNNCRAFT_GUILD, Map.of("name", guildName));
-        apiResponse.handleJsonObject(
-                json -> {
-                    if (!json.has("name")) {
-                        future.complete(Component.literal("Unable to check online members for " + guildName)
-                                .withStyle(ChatFormatting.RED));
-                        return;
-                    }
-
-                    MutableComponent response = Component.literal(
-                                    json.get("name").getAsString() + " ["
-                                            + json.get("prefix").getAsString() + "]")
-                            .withStyle(ChatFormatting.DARK_AQUA);
-
-                    response.append(Component.literal(" has ").withStyle(ChatFormatting.GRAY));
+            Guild guild;
 
-                    JsonObject guildMembers = json.getAsJsonObject("members");
+            try {
+                guild = completableFuture.get();
+            } catch (InterruptedException | ExecutionException e) {
+                completableFuture.whenComplete((result, throwable) ->
+                        McUtils.sendMessageToClient(Component.literal("Error trying to parse guild online members")
+                                .withStyle(ChatFormatting.RED)));
+                return;
+            }
```

Comment:
This will be blocking on the main thread, until the request completes. How about adding a `whenComplete` hook? (Obviously, for all cases).